utils/src/main/java/dev/arkbuilders/components/utils/PathExt.kt

```diff
@@ -0,0 +1,18 @@
+package dev.arkbuilders.components.utils
+
+import java.nio.file.Path
+
+fun Path.isParentOrSame(path: Path): Boolean {
+    return path.toString().startsWith(this.toString())
+}
+
+fun Path.isChildOrSame(path: Path): Boolean {
+    return this.toString().startsWith(path.toString())
+}
+
+fun Path.hasNestedOrParentalRoot(roots: Set<Path>): Boolean {
+    val hasNestedRoot = roots.indexOfFirst { path ->
+        this.isParentOrSame(path) || this.isChildOrSame(path)
+    } >= 0
```

Comment:
instead of `indexOfFirst {} >=0` you can use[ any](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/any.html)